### PR TITLE
Improved haddocks and explicit exports

### PR DESCRIPTION
**Pull request by solomon-b**
Adds more haddocks and explicit module exports. Also applies Brittany formatter (which we had previously agreed was the least disagreeable choice).

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -5,17 +5,19 @@ module Main where
 import           CofreeBot
 import           CofreeBot.Bot.Behaviors.Calculator.Language
 import           Control.Monad
+import           Control.Monad.Except           ( ExceptT
```
Comment (isovector) [inline app/Main.hs:8]: but it's still disagreeable af imo :upside_down_face:

File: src/CofreeBot/Bot.hs [hunk 2]
```diff
@@ -29,114 +69,27 @@ import           System.IO
...
-newtype Fix f = Fix { runFix :: f (Fix f) }
-
 -- | A 'Bot' maps from some input type 'i' and a state 's' to an
 -- output type 'o' and a state 's'
 type Bot :: KBot
 newtype Bot m s i o = Bot { runBot :: s -> i -> m (o, s) }
   deriving
     (Functor, Applicative, Monad, MonadState s, MonadReader i, MonadIO)
   via StateT s (ReaderT i m)
+instance Functor f => Profunctor (Bot f s) where
```
Comment (isovector) [inline src/CofreeBot/Bot.hs:84]: nit: no newline here?

**Review by isovector — APPROVED** [review]

File: app/Main.hs (hunk 1 above)
Comment (solomon-b) [reply app/Main.hs:8]: Haha I don't like it either! We definitely want some sort of code formatter just to keep things consistent but all our options seem to be bad. Maybe formolu with some customized options would be better?

File: src/CofreeBot/Bot.hs (hunk 2 above)
Comment (solomon-b) [reply src/CofreeBot/Bot.hs:84]: good catch

**Pull request merged**